samples/snippets/src/main/java/com/example/spanner/PgSpannerSample.java

```diff
@@ -0,0 +1,1603 @@
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Dialect;
+import com.google.cloud.spanner.Instance;
+import com.google.cloud.spanner.InstanceAdminClient;
+import com.google.cloud.spanner.InstanceId;
+import com.google.cloud.spanner.Key;
+import com.google.cloud.spanner.KeyRange;
+import com.google.cloud.spanner.KeySet;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.Options;
+import com.google.cloud.spanner.ReadOnlyTransaction;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerBatchUpdateException;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.SpannerExceptionFactory;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.cloud.spanner.Struct;
+import com.google.cloud.spanner.TimestampBound;
+import com.google.cloud.spanner.Value;
+import com.google.common.io.BaseEncoding;
+import com.google.longrunning.Operation;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.spanner.admin.database.v1.CreateBackupMetadata;
+import com.google.spanner.admin.database.v1.CreateDatabaseMetadata;
+import com.google.spanner.admin.database.v1.OptimizeRestoredDatabaseMetadata;
+import com.google.spanner.admin.database.v1.UpdateDatabaseDdlMetadata;
+import com.google.spanner.v1.ExecuteSqlRequest;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Example code for using the Cloud Spanner API. This example demonstrates all the common operations
+ * that can be done on Cloud Spanner. These are:
+ *
+ * <p>
+ *
+ * <ul>
+ *   <li>Creating a Cloud Spanner database.
+ *   <li>Writing, reading and executing SQL queries.
+ *   <li>Writing data using a read-write transaction.
+ *   <li>Using an index to read and execute SQL queries over data.
+ *   <li>Using commit timestamp for tracking when a record was last updated.
+ *   <li>Using Google API Extensions for Java to make thread-safe requests via long-running
+ *       operations. http://googleapis.github.io/gax-java/
+ * </ul>
+ */
+public class PgSpannerSample {
+  static final List<Singer> SINGERS =
+      Arrays.asList(
+          new Singer(1, "Marc", "Richards"),
+          new Singer(2, "Catalina", "Smith"),
+          new Singer(3, "Alice", "Trentor"),
+          new Singer(4, "Lea", "Martin"),
+          new Singer(5, "David", "Lomond"));
+  static final List<Album> ALBUMS =
+      Arrays.asList(
+          new Album(1, 1, "Total Junk"),
+          new Album(1, 2, "Go, Go, Go"),
+          new Album(2, 1, "Green"),
+          new Album(2, 2, "Forever Hold Your Peace"),
+          new Album(2, 3, "Terrified"));
+  // [START spanner_insert_data_with_timestamp_column]
```

Comment:
Also everywhere in this file: Make sure that we use unique tags for the PG samples.

Reply:
It will be better if we decide the format of the tag, will discuss this in today's call with JJ and do the necessary.